pom.xml

```diff
@@ -3300,6 +3300,13 @@
 
     <profile>
       <id>scala-2.12</id>
+      <properties>
+        <!-- 
+         SPARK-34774 Add this property to ensure change-scala-version.sh can replace the public `scala.version`
+         property correctly. 
+        -->
+        <scala.version>2.12.10</scala.version>
```

Comment:
This change will produce two`< scala.version >2.12.10</ scala.version >`, Should we extract a common property like `< scala.default.version >2.12.10</scala.default.version >` and reference it here as `<scala.version>${scala.default.version}</scala.version>` ?